desktop/core/src/desktop/js/catalog/optimizer/SqlAnalyzer.ts

```diff
@@ -98,6 +93,21 @@ export default class SqlAnalyzer implements Optimizer {
     });
   }
 
+  async checkMissingLimit(statement: string, dialect: string): Promise<boolean> {
+    const autocompleter = await sqlParserRepository.getAutocompleteParser(dialect);
+    const parsedStatement = autocompleter.parseSql(statement + ' ', '');
```

Comment:
Nit: Always good to wrap parseSql in a try catch. As not critical can just return false in the catch.

Reply:
I see, to follow-up in the next one:
- we need to switch to use a var and not a const to avoid clogging the catch scope?
- we silence the exception?

`
    let parsedStatement;
    try {
      parsedStatement = autocompleter.parseSql(statement + ' ', '');
    } catch (err) {
      return false;
    }
`